### PR TITLE
chore(trg-4.08): add draft for multi-platform builds

### PR DESCRIPTION
**Pull request by evegufy**
## Description

- add TRG for multi-platform (amd64 and arm64) builds of container images

this is especially beneficial for supporting use cases like tutorials and local playgrounds

- also, consolidate to one example workflow for TRG-4 section

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by evegufy** [comment]
Hi @almadigabor @carslen could you please do an initial review? Afterwards I'd communicate the PR on the mailing list. Thank you!

File: docs/release/trg-0/trg-4-08.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+
+    steps:
+      ...
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Set up QEMU
```
Comment (FaGru3n) [inline docs/release/trg-0/trg-4-08.md:39]: will this step also a must and is it nessesary for multi-build images? 
https://github.com/qemu/qemu

Or just additional information about the emulation and virtualisation,  then i would integrate a little bit more information on the "Why" section why we also want to have this step in the workflow files.

File: docs/release/trg-0/trg-4-08.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        uses: docker/setup-qemu-action@v3
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+           context: .
+           platforms: linux/amd64, linux/arm64
```
Comment (FaGru3n) [inline docs/release/trg-0/trg-4-08.md:46]: we already have two examples for build and push worflows 
https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-05
https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-01

maybe we could add the `platforms: linux/amd64, linux/arm64` directly within one of theese two trg´s and adding the mutli-platform info in there, or do you really want to have a separate mutli-platform TRG?

File: docs/release/trg-0/trg-4-08.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
+title: TRG 4.08 - Multi-platform
```
Comment (FaGru3n) [inline docs/release/trg-0/trg-4-08.md:2]:
```suggestion
title: TRG 4.07 - Multi-platform images
```
i guess next number for a TRG within the container section would be 4.07

**Review by FaGru3n — CHANGES_REQUESTED** [review]
Hi @evegufy, really good idea in my opinion.

But struggle a bit with a separate TRG for that and the separate step for QEMU... 

I think a additon to our registry TRG with "platforms: linux/amd64, linux/arm64" could also achive the goal to have multi-platform images

File: docs/release/trg-0/trg-4-08.md (hunk 3 above)
Comment (FaGru3n) [reply docs/release/trg-0/trg-4-08.md:2]: sorry missed that ther is already a 4.07

Comment (carslen) [reply docs/release/trg-0/trg-4-08.md:2]: There is a TRG 4.07 in TRG 0 section, so 4.08 shuld be fine for this proposal.

File: docs/release/trg-0/trg-4-08.md (hunk 2 above)
Comment (evegufy) [reply docs/release/trg-0/trg-4-08.md]: I consolidated to one example example workflow in TRG 4.01

File: docs/release/trg-0/trg-4-08.md (hunk 1 above)
Comment (evegufy) [reply docs/release/trg-0/trg-4-08.md]: I added a comment that those steps are necessary for multi-platfrom images

**Comment by evegufy** [comment]
> Hi @evegufy, really good idea in my opinion.
> 
> But struggle a bit with a separate TRG for that and the separate step for QEMU...
> 
> I think a additon to our registry TRG with "platforms: linux/amd64, linux/arm64" could also achive the goal to have multi-platform images

@FaGru3n thanks for reviewing and your feedback! I'd go for a separate TRG, as this entails a change to the workflows for most products, and having a separate TRG may be helpful for the product teams to keeping track of that change, also generally speaking most TRGs of 4.x are quite fine-granular, I don't think that this one stands out particularly.

**Review by almadigabor — APPROVED** [review]
I like this approach, also removing duplicates and referencing one workflow in 4.01 is much better IMO. I'm not sure about a separate TRG for multi-platform as it is a small topic but at least raises awareness among teams. I approve this from my side.

**Review by FaGru3n — APPROVED** [review]

**Comment by evegufy** [comment]
@almadigabor @FaGru3n I noticed that I probably need to improve on the implementation details https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/755/commits/08e32d9403ed82f4999d27faa22f9982a88edab4
@florianrusch-zf could you please also have a look at the PR?

**Review by FaGru3n — DISMISSED** [review]

**Review by florianrusch-zf — DISMISSED** [review]

**Comment by almadigabor** [comment]
Hey @evegufy, this was approved but the dependencies are out of date. Once they are updated, the PR can be merged I guess.

**Comment by evegufy** [comment]
Hi @florianrusch-zf @FaGru3n @carslen @almadigabor thanks for the initial feedback! I still improved the description for the implementation and wrote to the mailing list for review.

File: docs/release/trg-4/trg-4-08.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+## Why
+
+Providing container images that support multiple CPU architectures enhances flexibility, compatibility, and efficiency across hardware devices.
+
+This is especially beneficial for supporting use cases like tutorials and local playgrounds.
+
+## Description
+
+Container images **must** be built for **amd64 and arm64** CPU architectures.
```
Comment (paullatzelsperger) [inline docs/release/trg-4/trg-4-08.md:18]:
```suggestion
Container images **should** be built for **amd64 and arm64** CPU architectures.
```

I generally appreciate the idea of having multiplatform builds, but I don't see a point in making this mandatory. If it's easily supported, sure, why not. But converting a hypothetical image build that is non-multiplatform-capable into a multiplatform one could be a lengthy (and probably unnecessary) process, since with Rosetta emulation M[1|2|3] Macs can run Amd64 images.

TL;DR: swap **must** with **should** and i'll approve it.

Comment (evegufy) [reply docs/release/trg-4/trg-4-08.md]: @paullatzelsperger sure, I think that's fair for this topic, changed. thanks for reviewing!

File: docs/release/trg-4/trg-4-01.md [hunk 5]
```diff
@@ -92,48 +93,54 @@ jobs:
...
-           context: .
-           # Build image for verification purposes on every trigger event. Only push if event is not a PR
-           push: ${{ github.event_name != 'pull_request' }}
-           tags: ${{ steps.meta.outputs.tags }}
-           labels: ${{ steps.meta.outputs.labels }}
-
-       # https://github.com/peter-evans/dockerhub-description
-       # Important step to push image description to DockerHub 
+          context: .
+          # Needed to create multi-platfrom image
```
Comment (Siegfriedk) [inline docs/release/trg-4/trg-4-01.md:128]: Typo -> platfrom

File: docs/release/trg-4/trg-4-08.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+| Status | Created     | Post-History                              |
+|--------|-------------|-------------------------------------------|
+| Active | 01-Sep-2024 | Initial contribution                      |
+| Draft  | 20-Mar-2024 | Initial contribution                      |
+
+## Why
+
+Providing container images that support multiple CPU architectures enhances flexibility, compatibility, and efficiency across hardware devices.
+
+This is especially beneficial for supporting use cases like tutorials and local playgrounds.
```
Comment (Siegfriedk) [inline docs/release/trg-4/trg-4-08.md:14]: You could be a little bit more specific like mentioning ARM usage on MacOS and other runtimes like Amazon Gravition

**Review by Siegfriedk — APPROVED** [review]
Looks good, see comments

**Review by paullatzelsperger — APPROVED** [review]

**Review by florianrusch-zf — DISMISSED** [review]
No further comments from my side apart from those already mentioned

**Review by Siegfriedk — APPROVED** [review]

**Review by FaGru3n — APPROVED** [review]
nice work

**Review by SebastianBezold — APPROVED** [review]

**Review by Phil91 — APPROVED** [review]

**Comment by evegufy** [comment]
by now I count 7 approvals (just some aren't shown as they went stale) and I go ahead with merging
